crates/core/c8y_api/src/http_proxy.rs

```diff
@@ -144,14 +139,26 @@ impl C8yJwtTokenRetriever for C8yMqttJwtTokenRetriever {
         let token_smartrest =
             match tokio::time::timeout(Duration::from_secs(10), mqtt_con.received.next()).await {
                 Ok(Some(msg)) => msg.payload_str()?.to_string(),
-                Ok(None) => return Err(SMCumulocityMapperError::InvalidMqttMessage),
-                Err(_elapsed) => return Err(SMCumulocityMapperError::RequestTimeout),
+                Ok(None) => return Err(JwtError::NoJwtReceived),
+                Err(_elapsed) => return Err(JwtError::NoJwtReceived),
```

Comment:
Since both return the same error, why not as below?
```suggestion
               _ => return Err(JwtError::NoJwtReceived),          
```

Reply:
I try to avoid the `_` wild-card pattern as this somehow defeats the check of exhaustiveness done by the compiler. Say a new case is later added to the enum type (I agree this is unlikely here), then the compiler will fail to raise to my attention all the call sites using this wild-card.